baaas-decision-fleet-manager/src/main/java/org/kie/baaas/dfm/app/controller/AdminResource.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.baaas.dfm.app.controller;
+
+import java.util.List;
+
+import javax.annotation.security.RolesAllowed;
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.eclipse.microprofile.openapi.annotations.enums.SecuritySchemeType;
+import org.eclipse.microprofile.openapi.annotations.security.SecurityRequirement;
+import org.eclipse.microprofile.openapi.annotations.security.SecurityScheme;
+import org.eclipse.microprofile.openapi.annotations.security.SecuritySchemes;
+import org.kie.baaas.dfm.app.controller.modelmappers.DecisionMapper;
+import org.kie.baaas.dfm.app.manager.DecisionLifecycle;
+import org.kie.baaas.dfm.app.manager.DecisionLifecycleOrchestrator;
+import org.kie.baaas.dfm.app.model.Decision;
+import org.kie.baaas.dfm.app.model.DecisionVersion;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+@Path("/admin")
+@SecuritySchemes(value = {
+        @SecurityScheme(securitySchemeName = "bearer",
+                type = SecuritySchemeType.HTTP,
+                scheme = "Bearer")
+})
+@SecurityRequirement(name = "bearer")
+public class AdminResource {
+
+    private final Logger LOGGER = LoggerFactory.getLogger(AdminResource.class);
+
+    private final DecisionLifecycle decisionLifecycle;
+
+    private final DecisionMapper decisionMapper;
+
+    @Inject
+    public AdminResource(DecisionLifecycleOrchestrator decisionLifecycle,
+            DecisionMapper decisionMapper) {
+        requireNonNull(decisionLifecycle, "decisionLifecycle cannot be null");
+        requireNonNull(decisionMapper, "decisionMapper cannot be null");
+
+        this.decisionLifecycle = decisionLifecycle;
+        this.decisionMapper = decisionMapper;
+    }
+
+    @GET
+    @Path("/decisions")
+    @RolesAllowed({ "admin" })
+    public Response getDecisions() {
```

Comment:
Hi @robpblake , it's a fair point. I did not use `DecisionResponseList` because it is intended to contain information about the pagination and I thought it was not really needed to use pagination for admin (also because the consumer of the api would have anyway to retrieve all the decisions first and then take actions like deleting them, otherwise the data returned by the pagination would be messed up).

We might have some options: 
1) Do not use `DecisionResponseList` and return a list of decisions like I did
2) Use `DecisionResponseList` but without the pagination properties
3) use `DecisionResponseList` and populate the pagination data anyway (even if we do not implement pagination on that endpoint)
4) Implement the pagination on the endpoint so to be 100% consistent with the user endpoint. 

My very personal opinion is that being an "admin" endpoint we can consider it as internal, so we are free to change it when we want. At least until there is a real use case that requires pagination for example. Having said that, we might go for the option `2` or `3` and do not implement pagination for the time being, wdyt?

Reply:
> My very personal opinion is that being an "admin" endpoint we can consider it as internal, so we are free to change it when we want

Makes sense to me. 100% agree with you that there is no value in adding pagination to this Admin API right now. Let's go with it as it is 👍